GameEngine/GameState.cs

```diff
@@ -548,17 +548,30 @@ public Guid AddCharacter(Character character, Guid locationTrackingId)
             // If this is the player character, then keep track of the tracking id
             if (character is PlayerCharacter)
             {
-                Debug.Assert(PlayerTrackingId == Guid.Empty, $"A player character has already been set in the gamestate.");
-                PlayerTrackingId = character.TrackingId;
+                PlayerTrackingIds.Add(character.TrackingId);
             }
             Characters.Add(character.TrackingId, character);
             CharacterLocations[character.TrackingId] = locationTrackingId;
             return character.TrackingId;
         }
 
-        public Character GetPlayerCharacter()
+        public List<Character> GetPlayerCharacters()
         {
-            return GetCharacter(PlayerTrackingId);
+            return GetCharacters(PlayerTrackingIds);
+        }
+
+        public List<Character> GetCharacters(List<Guid> characterTrackingIds)
+        {
+            var characterList = new List<Character>();
```

Comment:
When you have 2 enumerable objects you can use Linq .Join to join them together and query based on the result.